lib/chef/resource/systemd_service.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Author:: Davide Cavalca (<dcavalca@fb.com>)
+# Copyright:: Copyright (c) 2015 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/service'
+require 'chef/provider/service/systemd'
+
+class Chef
+  class Resource
+    class SystemdService < Chef::Resource::Service
+      provides :service, os: "linux" do |node|
+        Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+      end
```

Comment:
Actually - don't we already do this?

from `service/debian.rb`

```
        provides :service, platform_family: 'debian' do |node|
          Chef::Platform::ServiceHelpers.service_resource_providers.include?(:debian)
        end
```

This seems to be following the existing pattern.

FTR, this is an actively blocker for us. We're monkeypatching this PR in now, but every day we go on adding uses of this is massively more painful if the upstream solution is different, so the sooner we can get an idea of what you think the proper solution is the better.

Also, i think other people are probably just manually making symlinks to /dev/null in /etc/ (which is what this does internally on `:mask`) which is... terrible and we should try to give them something more sane sooner.